### PR TITLE
47 - [feat] Custom Markers based on Renewable Energy Technology

### PR DESCRIPTION
**Pull request by deenasun**
### Description
[//]: # "[Required] Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
Updated `api/maps/AddMarkers.tsx`
- Included `technology` prop for the `MarkerInfoWindow` component so that each marker also receives information about the project's renewable energy technology type

Updated `api/maps/MarkerInfoWindow.tsx`
- Imported custom marker svgs
- Created map from renewable energy string to the corresponding svg
- Placed custom Image element inside the `AdvancedMarker` tags
- If the project's technology doesn't exist in the `technologyToPin` map, we show the default Pin icon

Imported custom marker svgs into `assets/Custom_Markers`

### Screenshots
[//]: # "Add any screenshots that will help your PR reviewer understand what’s going on in the PR!"
[//]: # "[Required] For frontend tasks: add screenshots of the layout and expected behavior (GIFs if you’re feeling extra fancy)!"
[//]: # "Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

## How to review
[//]: # "/[Required] Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

```
git fetch origin
git checkout 47-feat-custom-markers-based-on-renewable-energy-technology
pnpm run dev
```

Open up `localhost:3000`
You should see the markers populate the map! Feel free to navigate around the map and check that the clustering and hover functionality still works.

## Next steps
[//]: # "[Optional] - What has not yet been added to the PR, doesn’t yet have any functionality — essentially, what needs to be fixed or added."
[//]: # "[Optional] any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # "[Optional] insert links to any useful tutorials or documentation that you used for this PR!"
Documentation for Google Maps library: https://visgl.github.io/react-google-maps/docs/api-reference/components/advanced-marker

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @itsliterallymonique

### CODE REVIEW

File: api/maps/MarkerInfoWindow.tsx [hunk 1]
```diff
@@ -1,23 +1,44 @@
...
+import landbased_wind from '../../assets/Custom-Markers/landbased_wind.svg';
+import offshore_wind from '../../assets/Custom-Markers/offshore_wind.svg';
+import pumped_storage from '../../assets/Custom-Markers/pumped_storage.svg';
+import solarPower from '../../assets/Custom-Markers/solar_power.svg';
+
+const technologyToPin: Record<string, string> = {
+  'Energy Storage': energyStorage,
+  Geothermal: geothermal,
+  Hydroelectric: hydroelectric,
+  'Land Based Wind': landbased_wind,
```
Comment (itsliterallymonique) [inline api/maps/MarkerInfoWindow.tsx:22]: Land-Based Wind not Land Based

File: api/maps/MarkerInfoWindow.tsx [hunk 2]
```diff
@@ -1,23 +1,44 @@
...
+import geothermal from '../../assets/Custom-Markers/geothermal.svg';
+import hydroelectric from '../../assets/Custom-Markers/hydroelectric.svg';
+import landbased_wind from '../../assets/Custom-Markers/landbased_wind.svg';
+import offshore_wind from '../../assets/Custom-Markers/offshore_wind.svg';
+import pumped_storage from '../../assets/Custom-Markers/pumped_storage.svg';
+import solarPower from '../../assets/Custom-Markers/solar_power.svg';
+
+const technologyToPin: Record<string, string> = {
+  'Energy Storage': energyStorage,
+  Geothermal: geothermal,
```
Comment (itsliterallymonique) [inline api/maps/MarkerInfoWindow.tsx:20]: Why do some have the quotations marks `'' `and others don't?

**Review by itsliterallymonique — CHANGES_REQUESTED** [review]
Hey Deena! Amazing work :). Just address my 2 comments then you are good to squash and merge.

File: api/maps/MarkerInfoWindow.tsx (hunk 2 above)
Comment (deenasun) [reply api/maps/MarkerInfoWindow.tsx:20]: @itsliterallymonique I have no clue... I had the quotes around them but I think prettier takes them away for some of them for some reason if the keys are exact matches to the values (e.g. they have no extra spaces or anything)

Comment (itsliterallymonique) [reply api/maps/MarkerInfoWindow.tsx:20]: oh okay got it. Thanks deena!
